include/nbl/builtin/hlsl/shapes/beziers.hlsl

```diff
@@ -133,610 +198,205 @@ namespace shapes
-
-                    a[0]=lb;
-                    b[0]=ub;
-                }
+                qos = CsubPos + (B + A*t[2].x)*t[2].x;
+                dis = dot(qos,qos);
+                if( dis<res.x ) res = float2_t(dis,t[2].x );
                 
-                //further subdivide intervals to guarantee convergence of halley's method
-                //by using roots of further derivatives
-                float3 roots_snd_drv=float3(1e38,1e38,1e38);
-                int num_roots_snd_drv=solve_cubic(c2,roots_snd_drv);
-                sort_roots3(roots_snd_drv);
-
-                int num_roots_trd_drv=0;
-                float2 roots_trd_drv=float2(1e38,1e38);
-
-                if(num_roots_snd_drv!=3){
-                    num_roots_trd_drv=solve_quadric(c3,roots_trd_drv);
+                if(t[2].x != t[2].y)
+                {
+                    qos = CsubPos + (B + A*t[2].y)*t[2].y;
+                    dis = length(qos);
+                    if(dis < res.x) res = float2_t(dis, t[2].y);
                 }
 
-                for(int i=0;i<3;i++){
-                    if(i < num_roots){
-                        {
-                        for(int j=0;j<3;j+=2){
-                            if(j < num_roots_snd_drv){
-                                if(a[i] < roots_snd_drv[j] && b[i] > roots_snd_drv[j]){
-                                    if(eval_poly5(b0,b1,b2,b3,b4,roots_snd_drv[j]) > 0.){
-                                        b[i]=roots_snd_drv[j];
-                                    }
-                                    else{
-                                        a[i]=roots_snd_drv[j];
-                                    }
-                                }
-                            }
-                        }
-                        }
-                        
-                        for(int j=0;j<2;j++){
-                            if(j < num_roots_trd_drv){
-                                if(a[i] < roots_trd_drv[j] && b[i] > roots_trd_drv[j]){
-                                    if(eval_poly5(b0,b1,b2,b3,b4,roots_trd_drv[j]) > 0.){
-                                        b[i]=roots_trd_drv[j];
-                                    }
-                                    else{
-                                        a[i]=roots_trd_drv[j];
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
+                res.x = sqrt( res.x );
             }
+            
+            return res;
+        }
+        
+        // TODO: to be removed, keeping for reference
+        float_t signedDistance(float2_t pos, float_t thickness)
+        {
+            return abs(ud(pos)).x - thickness;
+        }
+        
+        template<typename Clipper>
+        float_t signedDistance2(float2_t pos, float_t thickness, ArcLengthPrecomputedValues preCompValues, Clipper clipper)
```

Comment:
remove `ArcLengthPrecomputedValues preCompValues` from inputs, it makes more sense to be in the clipper

Reply:
done